Demo/src/main/java/com/paypal/android/ui/paypalwebvault/PayPalWebVaultViewModel.kt

```diff
@@ -0,0 +1,143 @@
+package com.paypal.android.ui.paypalwebvault
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.paypal.android.api.model.PayPalPaymentToken
+import com.paypal.android.api.model.PayPalSetupToken
+import com.paypal.android.api.services.SDKSampleServerAPI
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.fraudprotection.PayPalDataCollector
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient
+import com.paypal.android.paypalwebpayments.PayPalWebVaultResult
+import com.paypal.android.paypalwebpayments.PayPalWebVaultListener
+import com.paypal.android.paypalwebpayments.PayPalWebVaultRequest
+import com.paypal.android.usecase.CreatePayPalPaymentTokenUseCase
+import com.paypal.android.usecase.CreatePayPalSetupTokenUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class PayPalWebVaultViewModel @Inject constructor(
+    val createPayPalSetupTokenUseCase: CreatePayPalSetupTokenUseCase,
+    val createPayPalPaymentTokenUseCase: CreatePayPalPaymentTokenUseCase,
+    val sdkSampleServerAPI: SDKSampleServerAPI
+) : ViewModel(), PayPalWebVaultListener {
+
+    companion object {
+        const val URL_SCHEME = "com.paypal.android.demo"
+    }
+
+    private val _uiState = MutableStateFlow(PayPalWebVaultUiState())
+    val uiState = _uiState.asStateFlow()
+
+    private lateinit var paypalClient: PayPalWebCheckoutClient
+    private lateinit var payPalDataCollector: PayPalDataCollector
+
```

Comment:
seems like you have PayPalDataCollector property with each viewModel. We should do that for iOS demo app as well.

Reply:
Yeah for now we're sending the client metadata ID along with calls to Capture / Authorize from the Demo app.